lib/api/LogManagerImpl.cpp

```diff
@@ -525,9 +538,12 @@ namespace MAT_NS_BEGIN
         m_context.SetCustomField(name, prop);
         {
             LOCKGUARD(m_dataInspectorGuard);
-            if (m_dataInspector)
+            for(const auto& inspector : m_dataInspectors)
             {
-                m_dataInspector->InspectSemanticContext(name, value, /*isGlobalContext: */ true, std::string{});
+                if(inspector->InspectorDescriptor().compare("PrivacyGuard") == 0)
```

Comment:
I don't believe we should have this check. All `IDataInspector`'s should have an `InspectSemanticContext` method and they should decide if they want to ignore it or inspect it. The logic should be an implementation details. This check is leaking the implementation detail into other components.

Reply:
yes I added that for both, but only removed it for one call of the `InspectSemanticContext`, will definitely just make the call and if it isn't necessary that call will be a no-op determined by that implementation of `IDataInspector`